target/scripts/helpers/utils.sh

```diff
@@ -17,9 +17,42 @@ function _escape_for_sed() {
 # Returns input after filtering out lines that are:
 # empty, white-space, comments (`#` as the first non-whitespace character)
 function _get_valid_lines_from_file() {
+  _convert_crlf_to_lf_if_necessary "${1}"
+  _append_final_newline_if_missing "${1}"
+
   grep --extended-regexp --invert-match "^\s*$|^\s*#" "${1}" || true
 }
 
+# This is to sanitize configs from users that unknowingly introduced CRLF:
+function _convert_crlf_to_lf_if_necessary() {
+  if [[ $(file "${1}") =~ 'CRLF' ]]; then
+    _log 'warn' "${1} contains CRLF line-endings."
+
+    if [[ -w "${1}" ]]; then
+      _log 'debug' 'Converting CRLF to LF'
+      sedfile -i 's|\r||g' "${1}"
+    fi
+  fi
+}
+
+# This is to sanitize configs from users that unknowingly removed the end-of-file LF:
+function _append_final_newline_if_missing() {
```

Comment:
This appears to be be causing the failures, which notably is making the change-detector unhappy. A config file is read with a missing newline and the modification to fix that sets off the change detector unintentionally.. (_something the earlier PR prevented with it's changes to `test.bats` user creation_)